csclient/params/params.go

```diff
@@ -260,6 +260,33 @@ type WhoAmIResponse struct {
 	Groups []string
 }
 
+// Resource describes a resource in the charm store.
+type Resource struct {
+	// Name identifies the resource.
+	Name string `json:"name"`
+
+	// Type is the name of the resource type.
+	Type string `json:"type"`
+
+	// Path is where the resource will be stored.
+	Path string `json:"path"`
+
+	// Description contains user-facing info about the resource.
+	Description string `json:"description,omitempty"`
+
+	// Origin is where the resource will come from.
+	Origin string `json:"origin"`
+
+	// Revision is the revision, if applicable.
+	Revision int `json:"revision"`
+
+	// Fingerprint is the SHA-384 checksum for the resource blob.
+	Fingerprint []byte `json:"fingerprint"`
+
+	// Size is the size of the resource, in bytes.
+	Size int64
+}
+
 const (
```

Comment:
They explicitly don't lowercase the json in this package... we should follow suit.  (also, IMO, it's better not to anyway).